src/block_gold.js

```diff
@@ -147,14 +147,19 @@ BlockGold.prototype.toHex = function (headersOnly) {
   return this.toBuffer(headersOnly).toString('hex')
 }
 
-BlockGold.prototype.checkProofOfWork = function (network) {
+BlockGold.prototype.checkProofOfWork = function (validateSolution, network) {
   var hash = this.getHash().reverse()
   var target = Block.calculateTarget(this.bits)
+  var validTarget = hash.compare(target) <= 0
```

Comment:
Better:
```
if (!validTarget) {
  return false
}

if (validateSolution) {
  ...
} else {
   return true
}
```

And btw, let's move `validateSolution` the the end of the param list and handle the `undefined` case so that it acts as a default param.

Reply:
I could argue the else is not needed, as there is return in the if, but it will make the code more readable if the else is present.